src/sdk/modules/validators/smartSessionValidator/smartSessionValidator.test.ts

```diff
@@ -0,0 +1,324 @@
+      smartSessionValidatorActions()
+    )
+
+    const enableSessionsResponse = await smartSessionNexusClient.enableSessions(
+      {
+        account: nexusClient.account,
+        sessionRequestedInfo: [sessionRequestedInfo]
+      }
+    )
+
+    expect(enableSessionsResponse.userOpHash).toBeDefined()
+    expect(enableSessionsResponse.permissionIds).toBeDefined()
+
+    const permissionIds = enableSessionsResponse.permissionIds
+    expect(permissionIds.length).toBe(1)
+    const permissionId = permissionIds[0]
+    cachedPermissionId = permissionId
+
+    const receipt = await nexusClient.waitForUserOperationReceipt({
+      hash: enableSessionsResponse.userOpHash
+    })
+
+    expect(receipt.success).toBe(true)
+
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: permissionId
+    })
+    expect(isEnabled).toBe(true)
+  }, 60000)
+
+  test("should make use of already enabled session (USE mode) to increment a counter using a session key", async () => {
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: cachedPermissionId
+    })
+    expect(isEnabled).toBe(true)
+
+    const smartSessionValidator = await toSmartSessionValidatorModule({
+      client: nexusClient.account.client as PublicClient,
+      initData: encodePacked(["address"], [eoaAccount.address]),
+      deInitData: "0x",
+      nexusAccountAddress: nexusClient.account.address,
+      activePermissionId: cachedPermissionId
+    })
+
+    const pubClient = nexusClient.account.client as PublicClient
+
+    const counterBefore = await pubClient.readContract({
+      address: TEST_CONTRACTS.Counter.address,
+      abi: CounterAbi,
+      functionName: "getNumber",
+      args: []
+    })
+
+    // helpful for out of range test
+    // await testClient.setNextBlockTimestamp({
+    //   timestamp: 9727001666n
+    // })
+
+    // set active validation module
+    nexusClient.account.setActiveValidationModule(smartSessionValidator)
+
+    const smartSessionNexusClient = nexusClient.extend(
+      smartSessionValidatorActions()
+    )
+
+    const userOpHash = await smartSessionNexusClient.useEnabledSession({
+      account: nexusClient.account,
+      actions: [
+        {
+          target: TEST_CONTRACTS.Counter.address,
+          value: 0n,
+          callData: encodeFunctionData({
+            abi: CounterAbi,
+            functionName: "incrementNumber",
+            args: []
+          })
+        }
+      ],
+      permissionId: cachedPermissionId
+    })
+
+    expect(userOpHash).toBeDefined()
+    const receipt = await nexusClient.waitForUserOperationReceipt({
+      hash: userOpHash
+    })
+    expect(receipt.success).toBe(true)
+
+    // // Make userop to increase counter
```

Comment:
comments

Reply:
why not?

Reply:
heh. I have no answer to that. They're fine!